### PR TITLE
test: Rewrite tests to use Spring Boot + devtools for reload

### PR DESCRIPTION
**Pull request by Artur-**
Jetty 10 does not support using scan combined with jetty:start.

This replaces tests need redeployment during the test with Spring Boot using Spring Boot devtools

### CODE REVIEW

**Comment by Artur-** [comment]
Now this should work...

File: flow-tests/test-redeployment/pom.xml [hunk 1]
```diff
@@ -32,6 +32,34 @@
...
+        </dependency>
+        <dependency>
+            <groupId>org.springframework.boot</groupId>
+            <artifactId>spring-boot-devtools</artifactId>
+            <version>${spring.boot.version}</version>
+        </dependency>
+        <dependency>
+            <groupId>com.vaadin</groupId>
+            <artifactId>vaadin-spring</artifactId>
+            <version>20.0-SNAPSHOT</version>
```
Comment (knoobie) [inline flow-tests/test-redeployment/pom.xml:60]: With this, the flow build depends on the spring repo resulting in a circular dependency.. is that worth it?

Comment (Artur-) [reply flow-tests/test-redeployment/pom.xml:60]: We should probably do https://github.com/vaadin/spring/issues/922

**Comment by Artur-** [comment]
Replacing this with #13010 as sonarcloud doesn't seem to understand what is not included in this PR anymore

**Pull request closed without merging**